### DISCUSSION TITLE
v6 - @storybook/addon-docs - code preview broken?

### DISCUSSION THREAD

**Discussion by wuifdesign** [storybookjs/storybook#11512] [General] [answered]
Hi,

i tried updating to v6.0.0-rc.1, but after the update the code blocks seam to act different than in v5. it now only displays the return value, and not the whole code used in the function (like in v5). Only showing the return value i loses important information i need in the code (variable declaration, ...).
It also displays components as `<[object Object] />`, loses linebrakes and rearrange attributes.

It also idisplays something like this:

```
<Title
  component={() => (
    <h1>This is the Title</h1> 
  )}
>
```

as

```
<Title
  component={function noRefCheck() {}}
>
```

i'm using react with typescript.

**Comment by shilman** [comment, marked as answer]
We've updated the code rendering in React to default to showing the virtual DOM rendered by the Story and this is probably the biggest area in need of stabilization during the RC. I'll try to provide a way to opt out of the new behavior in the next RC.

Tracked here #11526
